ipa-core/src/helpers/transport/in_memory/transport.rs

```diff
@@ -85,15 +85,14 @@ impl<I: TransportIdentity> InMemoryTransport<I> {
     /// out and processes it, the same way as query processor does. That will allow all tasks to be
     /// created in one place (driver). It does not affect the [`Transport`] interface,
     /// so I'll leave it as is for now.
-    fn listen<L: ListenerSetup<Identity = I>>(
+    fn listen(
         self: &Arc<Self>,
-        mut callbacks: L::Handler,
+        handler: Option<Box<dyn RequestHandler<Identity = I>>>,
```

Comment:
Is there a reason this should be optional in the dynamic case, but required in the static case? (Can we not substitute the panicking handler in the cases where this would otherwise be `None`?)

Reply:
I probably need to be consistent and either use `PanickingHandler` or `Option`. I decided that latter is better suited to indicate that there is no handler

Reply:
I agree that `Option` probably makes more sense that `PanickingHandler`, although if there's just a few test cases that want to omit a handler or something like that, then maybe better to use a stub in those cases than make everybody else deal with an `Option`.

In any case, I don't have a strong opinion either way, I was just curious why it was different than the static case.